server/app/services/export/QuestionJsonPresenter.java

```diff
@@ -0,0 +1,275 @@
+      return question.getApplicantQuestion().getContextualizedScalars().keySet().stream()
+          .filter(path -> !Scalar.getMetadataScalarKeys().contains(path.keyName()))
+          .collect(
+              toImmutableMap(
+                  path -> path,
+                  path ->
+                      question
+                          .getApplicantQuestion()
+                          .getApplicantData()
+                          .readAsString(path)
+                          .orElse("")));
+    }
+  }
+
+  class CurrencyJsonPresenter implements QuestionJsonPresenter<CurrencyQuestion> {
+    @Override
+    public ImmutableMap<Path, Double> getJsonEntries(CurrencyQuestion question) {
+      Path path =
+          question.getCurrencyPath().asApplicationPath().replacingLastSegment("currency_dollars");
+
+      if (question.getCurrencyValue().isPresent()) {
+        Long centsTotal = Long.valueOf(question.getCurrencyValue().get().getCents());
+
+        return ImmutableMap.of(path, centsTotal.doubleValue() / 100.0);
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+  }
+
+  class DateJsonPresenter implements QuestionJsonPresenter<DateQuestion> {
+    @Override
+    public ImmutableMap<Path, String> getJsonEntries(DateQuestion question) {
+      Path path = question.getDatePath().asApplicationPath();
+
+      if (question.getDateValue().isPresent()) {
+        LocalDate date = question.getDateValue().get();
+        return ImmutableMap.of(path, DateTimeFormatter.ISO_DATE.format(date));
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+  }
+
+  class EmptyJsonPresenter implements QuestionJsonPresenter<Question> {
+    @Override
+    public ImmutableMap<Path, ?> getJsonEntries(Question question) {
+      return ImmutableMap.of();
+    }
+  }
+
+  class NumberJsonPresenter implements QuestionJsonPresenter<NumberQuestion> {
+    @Override
+    public ImmutableMap<Path, Long> getJsonEntries(NumberQuestion question) {
+      Path path = question.getNumberPath().asApplicationPath();
+
+      if (question.getNumberValue().isPresent()) {
+        return ImmutableMap.of(path, question.getNumberValue().get());
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+  }
+
+  class PhoneJsonPresenter implements QuestionJsonPresenter<PhoneQuestion> {
+
+    private static final PhoneNumberUtil PHONE_NUMBER_UTIL = PhoneNumberUtil.getInstance();
+
+    @Override
+    public ImmutableMap<Path, String> getJsonEntries(PhoneQuestion question) {
+      Path path = question.getPhoneNumberPath().asApplicationPath();
+
+      if (question.getPhoneNumberValue().isPresent()
+          && question.getCountryCodeValue().isPresent()) {
+        String formattedPhone =
+            getFormattedPhoneNumber(
+                question.getPhoneNumberValue().get(), question.getCountryCodeValue().get());
+        return ImmutableMap.of(path, formattedPhone);
+      } else {
+        return ImmutableMap.of();
+      }
+    }
```

Comment:
Above code is taken from `case PHONE`:

https://github.com/civiform/civiform/blob/358ad3dbb6e67aacb91afb3ede60e32ac281ba26/server/app/services/export/JsonExporter.java#L130-L219